### PR TITLE
Provide HTTP-Header information on demand

### PR DESCRIPTION
**Pull request by desperateCoder**
Hi @David-Development and @tobiasKaminsky,

here you go!

### CODE REVIEW

File: src/main/java/com/nextcloud/android/sso/api/NextcloudAPI.java [hunk 1]
```diff
@@ -127,6 +147,7 @@ public Response performRequestV2(final @NonNull Type type, NextcloudRequest requ
         Reader targetReader = new InputStreamReader(response.getBody());
 
         if (type != Void.class) {
+            //FIXME: @original-devs: this will never work (ClassCastException), unless type is Response.
```
Comment (David-Development) [inline src/main/java/com/nextcloud/android/sso/api/NextcloudAPI.java:150]: mhm.. what exactly will never work? 🤔 The gson parsing?

**Comment by desperateCoder** [comment]
Yep, after your refactoring everything still seems to work as expected. From my point of view this is safe to merge.

The DCO-error seems to be your commit btw @David-Development , i already had amended mine.

**Comment by David-Development** [comment]
@desperateCoder Nice, looks good to me, too! Thank you!

@tobiasKaminsky feel free to merge 🙂

**Comment by David-Development** [comment]
Should we merge this one? 😂 I thought one of you guys is going to merge it since I approved everything 🙈

**Comment by stefan-niedermann** [comment]
Neither @desperateCoder nor me have merge permissions in this repository. If one grants me the permissions, i will happily merge it...

**Comment by David-Development** [comment]
@stefan-niedermann Oh.. good to know! Maybe @tobiasKaminsky can grant them to you? 

In the meantime I'll merge this into master

**Pull request merged**